common/src/main/java/com/wynntils/services/mapdata/providers/json/JsonProvider.java

```diff
@@ -180,20 +160,60 @@ public void onChange(Consumer<MapDataProvidedType> callback) {
 
     @Override
     public void reloadData() {
-        // FIXME: To be implemented if needed (when the first json provider is added)
+        // Json providers do not need to reload data, as the whole object is reloaded by the owner service instead
+    }
+
+    public static final class JsonProviderSerializer implements JsonDeserializer<JsonProvider> {
+        @Override
+        public JsonProvider deserialize(JsonElement json, Type type, JsonDeserializationContext context)
+                throws JsonParseException {
+            JsonObject jsonProvider = json.getAsJsonObject();
+
+            int version = JsonUtils.getNullableJsonInt(jsonProvider, "version");
+            if (version == 0) {
+                throw new JsonParseException("Missing or incorrect version field in json provider.");
+            }
```

Comment:
Maybe we should also verify that the version number matches our expectation, like `1`?